libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -0,0 +1,1033 @@
+            let value: any;
+            let error: string;
+            let args: any[];
+            ({args, error} = FunctionUtils.evaluateChildren(expression, state, options, verify));
+            if (!error) {
+                try {
+                    ({value, error} = func(args));
+                } catch (e) {
+                    error = e.message;
+                }
+            }
+
+            return {value, error};
+        };
+    }
+
+    /**
+     * Generate an expression delegate that applies function on the accumulated value after verifying all children.
+     * @param func Function to apply.
+     * @param verify Function to check each arg for validity.
+     * @returns Delegate for evaluating an expression.
+     */
+    public static applySequence(func: (arg0: any[]) => any, verify?: VerifyExpression): EvaluateExpressionDelegate {
+        return FunctionUtils.apply(
+            (args: any[]): any => {
+                const binaryArgs: any[] = [undefined, undefined];
+                let soFar: any = args[0];
+                for (let i = 1; i < args.length; i++) {
+                    binaryArgs[0] = soFar;
+                    binaryArgs[1] = args[i];
+                    soFar = func(binaryArgs);
+                }
+
+                return soFar;
+            },
+            verify
+        );
+    }
+
+    /**
+     * Generate an expression delegate that applies function on the accumulated value after verifying all children.
+     * @param func Function to apply.
+     * @param verify Function to check each arg for validity.
+     * @returns Delegate for evaluating an expression.
+     */
+    public static applySequenceWithError(func: (arg0: any[]) => any, verify?: VerifyExpression): EvaluateExpressionDelegate {
+        return FunctionUtils.applyWithError(
+            (args: any[]): any => {
+                const binaryArgs: any[] = [undefined, undefined];
+                let soFar: any = args[0];
+                let value: any;
+                let error: string;
+                for (let i = 1; i < args.length; i++) {
+                    binaryArgs[0] = soFar;
+                    binaryArgs[1] = args[i];
+                    ({value, error} = func(binaryArgs));
+                    if (error) {
+                        return {value, error};
+                    } else {
+                        soFar = value;
+                    }
+
+                }
+
+                return {value: soFar, error: undefined};
+            },
+            verify
+        );
+    }
+
+    /**
+     * Lookup a property in IDictionary, JObject or through reflection.
+     * @param instance Instance with property.
+     * @param property Property to lookup.
+     * @returns Value and error information if any.
+     */
+    public static accessProperty(instance: any, property: string): {value: any; error: string} {
+        // NOTE: This returns null rather than an error if property is not present
```

Comment:
>// NOTE: This returns null rather than an error if property is not present [](start = 8, length = 74)

I don't understand these comments.  How would this function ever return null?  Typescript enforces the structure and you always return it.

Reply:
it is a typo. If the instance is undefined, the accessor would return undefined directly.